packages/lesswrong/components/posts/AllPostsPage.tsx

```diff
@@ -28,166 +42,96 @@ export const timeframes = {
-            limit: 50
-          }}
-          dimWhenLoading={showSettings}
-        />
-      </AnalyticsContext>
-    }
-
-    const numTimeBlocks = timeframeToNumTimeBlocks[currentTimeframe as TimeframeType]
-    const timeBlock = timeframeToTimeBlock[currentTimeframe as TimeframeType]
-    
-    let postListParameters: PostsViewTerms = {
-      view: 'timeframe',
-      ...baseTerms
-    }
-
-    if (parseInt(query.limit)) {
-      postListParameters.limit = parseInt(query.limit)
-    }
-    
-    return <div>
-      <AnalyticsContext
-        listContext={"allPostsPage"}
-        terms={postListParameters}
-        capturePostItemOnMount
-      >
-        {/* Allow unhiding posts from all posts menu to allow recovery of hiding the wrong post*/}
-        <AllowHidingFrontPagePostsContext.Provider value={true}>
-          <PostsTimeframeList
-            // TODO: this doesn't seem to be guaranteed, actually?  Since it can come from an unsanitized query param...
-            timeframe={currentTimeframe as TimeframeType}
-            postListParameters={postListParameters}
-            numTimeBlocks={numTimeBlocks}
-            dimWhenLoading={showSettings}
-            after={query.after || getAfterDefault({numTimeBlocks, timeBlock, timezone, before: query.before})}
-            before={query.before  || getBeforeDefault({timeBlock, timezone, after: query.after})}
-            reverse={query.reverse === "true"}
-            displayShortform={query.includeShortform !== "false"}
+  }, [showSettings, captureEvent, currentUser, updateCurrentUser]);
+
+  const currentTimeframe = query.timeframe || currentUser?.allPostsTimeframe || 'daily';
+  const currentSorting = query.sortedBy    || currentUser?.allPostsSorting   || 'magic';
+  const currentFilter = query.filter       || currentUser?.allPostsFilter    || 'all';
+  const currentShowLowKarma = (parseInt(query.karmaThreshold) === MAX_LOW_KARMA_THRESHOLD) ||
+    currentUser?.allPostsShowLowKarma || false;
+  const currentIncludeEvents = (query.includeEvents === 'true') || currentUser?.allPostsIncludeEvents || false;
+  const currentHideCommunity = (query.hideCommunity === 'true') || currentUser?.allPostsHideCommunity || false;
+
+  const {
+    SingleColumnSection, SectionTitle, SortButton, PostsListSettings, HeadTags,
+    AllPostsList,
+  } = Components;
+
+  return (
+    <>
+      <HeadTags description={description} />
+      <AnalyticsContext pageContext="allPostsPage">
+        <SingleColumnSection>
+          <Tooltip
+            title={`${showSettings ? "Hide": "Show"} options for sorting and filtering`}
+            placement="top-end"
+          >
+            <div className={classes.title} onClick={toggleSettings}>
+              <SectionTitle title={preferredHeadingCase("All Posts")}>
+                <SortButton label={formatSort(currentSorting)} />
+              </SectionTitle>
+            </div>
+          </Tooltip>
+          {!showSettings && <hr className={classes.divider} />}
```

Comment:
I feel like it's better to forum-gate this element rather than forum-gate the styles with a `display: none`, since then we're still including an extra element